tests/schemas/builder.py

```diff
@@ -48,8 +58,16 @@ def field_to_spec(field):
     # Handle nested/meta/special types
     if isinstance(field, CheckedConstant):
         spec["value"] = field.constant
-    if isinstance(field, EnumCheckedConstant):
+    elif isinstance(field, EnumCheckedConstant):
         spec["value"] = field.constant.value
+        spec["enum_type"] = type(field.constant).__name__
```

Comment:
Should this field be called `enum_constant_type` so it's not confused with the `enum_type` of enum fields?

Reply:
I think it's fine given 1) it's the same kind of data that goes in both case and 2) the `spec["type"]` field already discriminate (the value being either `EnumCheckedConstant` or `MyEnumField`)